sui/src/sui_commands.rs

```diff
@@ -264,12 +278,36 @@ pub struct SuiNetwork {
 }
 
 impl SuiNetwork {
-    pub async fn start(config: &NetworkConfig) -> Result<Self, anyhow::Error> {
+    pub async fn start(
+        config: &NetworkConfig,
+        key_pairs: Vec<&KeyPair>,
+    ) -> Result<Self, anyhow::Error> {
         if config.authorities.is_empty() {
             return Err(anyhow!(
                 "No authority configured for the network, please run genesis."
             ));
         }
+        if config.authorities.len() != key_pairs.len() {
+            return Err(anyhow!(
+                "Num of authorities does not match num of key_pairs."
+            ));
+        }
+
+        // Align key_pairs' order with config.authorities'
+        let mut ordered_key_pairs = Vec::with_capacity(key_pairs.len());
+        for authority in &config.authorities {
+            let key_pair = key_pairs
+                .iter()
+                .find(|p| SuiAddress::from(p.public_key_bytes()) == authority.address)
+                .ok_or_else(|| {
+                    anyhow!(
+                        "Can't find keypair for authority {:?}",
+                        authority.public_key,
+                    )
+                })?;
```

Comment:
This reordering seems to be O(N^2). It would be great to use a `HashMap` from public keys to key pair to make it O(N).